qiskit_experiments/data_processing/nodes.py

```diff
@@ -488,85 +428,71 @@ def __init__(
                     "Prior for probability node must be a float or pair of floats."
                 )
             self._alpha_prior = list(alpha_prior)
+
         super().__init__(validate)
 
-    def _format_data(self, datum: dict, error: Optional[Any] = None) -> Tuple[dict, Any]:
+    def _format_data(self, data: np.ndarray) -> np.ndarray:
         """
         Checks that the given data has a counts format.
 
         Args:
-            datum: An instance of data the should be a dict with bit strings as keys
-                and counts as values.
+            data: An object data array containing count dictionaries.
 
         Returns:
-            The datum as given.
+            The ``data`` as given.
 
         Raises:
             DataProcessorError: if the data is not a counts dict or a list of counts dicts.
         """
-        if self._validate:
-
-            if isinstance(datum, dict):
-                data = [datum]
-            elif isinstance(datum, list):
-                data = datum
-            else:
-                raise DataProcessorError(f"Datum must be dict or list, received {type(datum)}.")
+        valid_count_type = int, float, np.integer
 
-            for datum_ in data:
-                if not isinstance(datum_, dict):
+        if self._validate:
+            for datum in data:
+                if not isinstance(datum, dict):
                     raise DataProcessorError(
-                        f"Given counts datum {datum_} to "
-                        f"{self.__class__.__name__} is not a valid count format."
+                        f"Data entry must be dictionary of counts, received {type(datum)}."
                     )
-
-                for bit_str, count in datum_.items():
+                for bit_str, count in datum.items():
                     if not isinstance(bit_str, str):
                         raise DataProcessorError(
-                            f"Key {bit_str} is not a valid count key in{self.__class__.__name__}."
+                            f"Key {bit_str} is not a valid count key in {self.__class__.__name__}."
                         )
-
-                    if not isinstance(count, (int, float, np.integer)):
+                    if isinstance(count, Variable):
+                        # This code eliminates variance in count values.
+                        # This variance may be generated by readout error mitigation when
+                        # A-matrix is computed with some error.
+                        datum[bit_str] = count.nominal_value
+                    if not isinstance(count, valid_count_type):
                         raise DataProcessorError(
-                            f"Count {bit_str} is not a valid count value in {self.__class__.__name__}."
+                            f"Count {bit_str} is not a valid value in {self.__class__.__name__}."
                         )
 
-        return datum, None
+        # This is bare ndarray. No error will be attached to data.
```

Comment:
Can't we get an error from this?

Reply:
We can include error in the count values and compute probability error from that. This is the case that we compute count dict with error mitigation matrix including sampling error. However, this should be different node. Note that `Probability` node computes sampling error from binominal distribution, and error should be determined by its frequency and total counts (not uncertainty of counts).

Reply:
@chriseclectic any thoughts?

Reply:
The frequencies in counts shouldn't have error since they need to be a discrete distribution. The error should only be in the quantities (probabilities / quasi-probabilities / expvals) computed from them (or should be in the estimation of the distribution they were sampled from like is done internally in prob node).

For error mitigation you shouldn't mitigate counts like was previously done, in newer mitigation you either get back mitigated quasi-probabilities or expval which can both have error.

Reply:
Thanks Chris, that makes sense. I updated the validation so that it only accepts integer values. Float, i.e. mitigated, count and its uncertainty should be handled by https://github.com/Qiskit/qiskit-terra/pull/6485 rather than this node.

Done in https://github.com/Qiskit/qiskit-experiments/pull/481/commits/45003dc5f78c90c3ce5b10238bf6f079b9ee50f0